### ISSUE TITLE
Support for Synapse Workspace Managed Identity "Allow pipelines" permission

### ISSUE THREAD

**Issue by markhoiland** [hashicorp/terraform-provider-azurerm#10137]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The Synapse Managed Identity configuration includes a flag for allowing pipelines to use the system-assigned managed identity to access SQL pools that are configured within the workspace.  This is a critical functionality for workspace functionality.

### New or Affected Resource(s)

* azurerm_synapse_workspace

### Potential Terraform Configuration

```hcl
resource "azurerm_synapse_workspace" "example" {
  name                                 = "example"
  resource_group_name                  = azurerm_resource_group.example.name
  location                             = azurerm_resource_group.example.location
  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
  sql_administrator_login              = "sqladminuser"
  sql_administrator_login_password     = "H@Sh1CoR3!"
  allow_managed_identity_sql_control   = true

  aad_admin {
    login     = "AzureAD Admin"
    object_id = "00000000-0000-0000-0000-000000000000"
    tenant_id = "00000000-0000-0000-0000-000000000000"
  }
}
```

### References

* https://docs.microsoft.com/en-us/rest/api/synapse/workspacemanagedidentitysqlcontrolsettings/createorupdate
* https://docs.microsoft.com/en-us/azure/synapse-analytics/security/how-to-grant-workspace-managed-identity-permissions

**Comment by allantargino** [comment]
Hi @markhoiland,
This was fixed by #10033, and the property is called `sql_identity_control_enabled`.

**Comment by aristosvo** [comment]
Closed as fixed by #10033
